### PR TITLE
Adjust windows according to their desired increment.  Notify windows when their size changes

### PR DESCRIPTION
**Pull request by okratitan**
This follows the icccm spec and also fixes applications like terminology.

### CODE REVIEW

File: wm/property.go [hunk 1]
```diff
@@ -103,6 +104,34 @@ func windowMinSize(x *xgbutil.XUtil, win xproto.Window) (uint, uint) {
 	return 0, 0
 }
 
+func windowApplyIncrement(x *xgbutil.XUtil, win xproto.Window, width uint16, height uint16) (uint16, uint16) {
```
Comment (andydotxyz) [inline wm/property.go]: This is not actually applying anything - it is returning a value so verbs like calculate, round or similar would be clearer.
Perhaps even windowSizeWithIncrement?

Comment (okratitan) [reply wm/property.go]: updated to be windowSizeWithIncrement

**Comment by okratitan** [comment]
Okay the second commit further improves our window handling.  It follows requested resize increments, and also alerts clients to their geometry according to the icccm spec so that they now know where they are located and where to place their sub menus/windows.  This also fixes the reframing on fyndesk_runner reload to where the windows are no longer offset but kept in their same position.

You will now see gnome-terminal, lxterminal, terminology, goland, etc..., etc... all working the way they should.

**Comment by andydotxyz** [comment]
Very close. The reparent on restart is a great fix.
However when you drag a window after a restart it then jumps to the offset position it would have been at before.

File: wm/frame.go [hunk 2]
```diff
@@ -397,7 +411,10 @@ func (f *frame) updateTitle() {
 }
 
 func (f *frame) borderWidth() uint16 {
-	return f.client.wm.scaleToPixels(wmTheme.BorderWidth)
+	if f.client.Decorated() {
```
Comment (andydotxyz) [inline wm/frame.go]: This logic seems the wrong way round. The main content of a function should be at the top level.
So if !decorated { return 0 } would read more logically

**Pull request merged**